### PR TITLE
[DOC] Added examples for arrayContent[Will|Did]Change

### PR DESCRIPTION
**Pull request by mukilane**
For #18228 

I've added a basic example for both arrayContentWillChange & arrayContentDidChange. 

I've also added some twiddles to support them.
[arrayContentWillChange](https://ember-twiddle.com/de4000d2daff26396c1b6525d341d307?openFiles=templates.components.my-component.hbs%2Ctemplates.components.my-component.hbs) and [arrayContentDidChange](https://ember-twiddle.com/67cfb2ec4edc9b3aea8cb664bba8ea49?openFiles=templates.components.my-component.hbs%2Ctemplates.components.my-component.hbs)

Would love to have some inputs on how to improve them.

### CODE REVIEW

File: packages/@ember/-internals/runtime/lib/mixins/array.js [hunk 1]
```diff
@@ -491,6 +491,49 @@ const ArrayMixin = Mixin.create(Enumerable, {
     invalidate any related properties. Pass the starting index of the change
     as well as a delta of the amounts to change.
 
+    ```javascript
```
Comment (locks) [inline packages/@ember/-internals/runtime/lib/mixins/array.js]: Can you give this component a path?

    ```app/templates/show-post.js

Or something like that

Comment (locks) [inline packages/@ember/-internals/runtime/lib/mixins/array.js]: You are missing the imports in this code block.

File: packages/@ember/-internals/runtime/lib/mixins/array.js [hunk 2]
```diff
@@ -491,6 +491,49 @@ const ArrayMixin = Mixin.create(Enumerable, {
...
+    const Post = EmberObject.extend({
+      body: '',
+      save() {}
+    })
+
+    export default Component.extend({
+
```
Comment (locks) [inline packages/@ember/-internals/runtime/lib/mixins/array.js]:
```suggestion
```

File: packages/@ember/-internals/runtime/lib/mixins/array.js [hunk 3]
```diff
@@ -491,6 +491,49 @@ const ArrayMixin = Mixin.create(Enumerable, {
...
+      attemptsToModify: 0,
+      successfulModifications: 0,
+      posts: null,
+
+      init() {
+        this._super(...arguments);
```
Comment (locks) [inline packages/@ember/-internals/runtime/lib/mixins/array.js]:
```suggestion
        this._super(...arguments);

```

File: packages/@ember/-internals/runtime/lib/mixins/array.js [hunk 4]
```diff
@@ -491,6 +491,49 @@ const ArrayMixin = Mixin.create(Enumerable, {
...
+          didChange() {
+            this.incrementProperty('successfulModifications');
+          }
+        });
+      },
+
+      actions: {
+        editPost(post, newContent) {
+          let oldContent = post.body;
+          this.posts.arrayContentWillChange(); // attemptsToModify = 1
```
Comment (locks) [inline packages/@ember/-internals/runtime/lib/mixins/array.js]: You are not passing any of the two required params to `arrayContentWillChange`, can you update?

**Review by locks — CHANGES_REQUESTED** [review]
Left some remarks inline, thanks for the help!

**Comment by mukilane** [comment]
@locks , I've updated them.

**Comment by mukilane** [comment]
@locks Is any more changes required ?

**Review by locks — APPROVED** [review]
Looking good, let's get it in!

**Pull request merged**